addons/website/models/website_menu.py

```diff
@@ -26,6 +26,8 @@ def _default_sequence(self):
     child_id = fields.One2many('website.menu', 'parent_id', string='Child Menus')
     parent_path = fields.Char(index=True)
     is_visible = fields.Boolean(compute='_compute_visible', string='Is Visible')
+    group_ids = fields.Many2many('res.groups', string='Visible Groups',
```

Comment:
Rare case but still we can consider this as Improvement as we are adding new field.
Goog to add relation explicitly so we can prevent situation like this (https://github.com/odoo/odoo/pull/29376) in future.